e2e/Tests/provisioning/ProvisioningE2ETests.cs

```diff
@@ -686,12 +687,12 @@ public async Task DPS_Registration_AmqpWs_X509_IndividualEnrollment_InvalidGloba
 
             try
             {
-                using var overallCts = new CancellationTokenSource(PassingTimeoutMilliseconds);
+                using var overallCts = new CancellationTokenSource(RegisterWithRetryTimeoutMilliseconds);
 
                 // Trying to register simultaneously can cause conflicts (409). Retry in those scenarios to succeed.
                 while (true)
                 {
-                    using var attemptCts = new CancellationTokenSource(FailingTimeoutMilliseconds);
+                    using var attemptCts = new CancellationTokenSource(RegisterTimeoutMilliseconds);
                     using var opCts = CancellationTokenSource.CreateLinkedTokenSource(overallCts.Token, attemptCts.Token);
```

Comment:
since opCts will cancel when either overallCts or attemptCts cancels, it is limited by the cancellation token with the shortest timeout. Then, since we own both cancellation tokens, why not pass the shortest duration one directly? Is someone else signalling them for cancellation?

Reply:
> why not pass the shortest duration one directly?

Is there a way to compare cancellation tokens like that? If so, we could do that. If not, then this is the only way I know to cancel an operation when one or more tokens are canceled

Reply:
> Is someone else signalling them for cancellation?

No, just the timeout value it is constructed with will signal each of these tokens

Reply:
> since opCts will cancel when either overallCts or attemptCts cancels, it is limited by the cancellation token with the shortest timeout. Then, since we own both cancellation tokens, why not pass the shortest duration one directly? Is someone else signalling them for cancellation?

Because we can still retry if the original response failed to get back to our client for some reason. If we just have 1 cancellation token then you get one shot and that is it.